include/uv.h

```diff
@@ -1899,6 +1899,9 @@ UV_EXTERN int uv_thread_getcpu(void);
 UV_EXTERN uv_thread_t uv_thread_self(void);
 UV_EXTERN int uv_thread_join(uv_thread_t *tid);
 UV_EXTERN int uv_thread_equal(const uv_thread_t* t1, const uv_thread_t* t2);
+UV_EXTERN int uv_thread_setname(const char* name);
+UV_EXTERN int uv_thread_getname(uv_thread_t* tid, char* name, size_t size);
```

Comment:
All similar APIs _except_ `uv_get_process_title` get a pointer to the size where they write the length.

Reply:
I'm not saying we should, just pointing out an API quirk. We cannot change the other API but we can think about it for this one ;-)

Reply:
I think it could work. Another option could be that considering there's a well defined limit: `UV_PTHREAD_MAX_NAMELEN_NP`, how about we make that value public and document that the buffer passed should have that length (or we could do both)

Reply:
That may be painting ourselves in a corner because what if we grow support for a platform where the limit is not so well-defined?

Reply:
Fair enough. I'll keep the API as it is.